Library/Homebrew/github_runner_matrix.rb

```diff
@@ -122,13 +123,30 @@ def create_runner(platform, arch, spec, macos_version = nil)
 
   NEWEST_HOMEBREW_CORE_MACOS_RUNNER = :sequoia
   OLDEST_HOMEBREW_CORE_MACOS_RUNNER = :ventura
-  NEWEST_HOMEBREW_CORE_INTEL_MACOS_RUNNER = :sonoma
+  NEWEST_DEFAULT_HOMEBREW_CORE_INTEL_MACOS_RUNNER = :sonoma
 
   sig { params(macos_version: MacOSVersion).returns(T::Boolean) }
   def runner_enabled?(macos_version)
     macos_version <= NEWEST_HOMEBREW_CORE_MACOS_RUNNER && macos_version >= OLDEST_HOMEBREW_CORE_MACOS_RUNNER
   end
 
+  NEW_INTEL_MACOS_MUST_BUILD_FORMULAE = %w[pkg-config pkgconf].freeze
+
+  sig { returns(T::Boolean) }
+  def deploy_new_x86_64_runner?
+    return true if @testing_formulae.any? { |f| NEW_INTEL_MACOS_MUST_BUILD_FORMULAE.include?(f.name) }
+    return true if @testing_formulae.any? { |f| f.formula.class.pour_bottle_only_if == :clt_installed }
+
+    Formula.all.any? do |formula|
```

Comment:
You probably need to pass `eval_all` here

Reply:
Yep, will add it when I get around to updating the calls in `homebrew/core`